src/components/attendance/attendance.controller.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  UseGuards,
+  UseFilters,
+} from '@nestjs/common';
+import { AttendanceService } from './attendance.service';
+import { CreateAttendanceDto } from './dto/create-attendance.dto';
+import { UpdateAttendanceDto } from './dto/update-attendance.dto';
+import { AttendanceEntity } from './entity/attendance.entity';
+import { RoleGuard } from '../../guards/authorization.guard';
+import { Roles } from '../../guards/guard.role.decorator';
+import { UserRole } from '../user/entity/user.entity';
+import { HttpExceptionFilter } from 'src/exception/http-exception.filter';
+
+@Controller('attendance')
+@Roles(UserRole.Admin, UserRole.StaffMember)
+@UseGuards(RoleGuard)
+@UseFilters(HttpExceptionFilter)
+export class AttendanceController {
+  constructor(private readonly attendanceService: AttendanceService) {}
+
```

Comment:
Follow REST api, No need to write get, create, delete, update keywords in route name, API decorator isself represents use case of APIs. e.g. Just use @Post() for add attendance.
Change everywhere in this file.

Reply:
@Post('/add-attendance')
  create(
    @Body() createAttendanceDto: CreateAttendanceDto,
  ): Promise<AttendanceEntity> {
    return this.attendanceService.create(createAttendanceDto);
  }
  
  I have done this way means decorator is added can u explain me what you are telling?